docs/addons/opa-gatekeeper.md

```diff
@@ -0,0 +1,195 @@
+
+The policy should look like the following: 
+
+```yaml
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sallowedrepos
+  annotations:
+    description: Requires container images to begin with a repo string from a specified
+      list.
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sAllowedRepos
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          type: object
+          properties:
+            repos:
+              type: array
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sallowedrepos
+        violation[{"msg": msg}] {
+          container := input.review.object.spec.containers[_]
+          satisfied := [good | repo = input.parameters.repos[_] ; good = startswith(container.image, repo)]
+          not any(satisfied)
+          msg := sprintf("container <%v> has an invalid image repo <%v>, allowed repos are %v", [container.name, container.image, input.parameters.repos])
+        }
+        violation[{"msg": msg}] {
+          container := input.review.object.spec.initContainers[_]
+          satisfied := [good | repo = input.parameters.repos[_] ; good = startswith(container.image, repo)]
+          not any(satisfied)
+          msg := sprintf("container <%v> has an invalid image repo <%v>, allowed repos are %v", [container.name, container.image, input.parameters.repos])
+        }
+```
+
+All the Gatekeeper policy examples can be found under the examples directory in the SSP repository. If you go into the examples directory and go into the gatekeeper-library folder, we can apply this to our cluster by running the 
+```bash
+kubectl apply -f /examples/gatekeeper-library/library/general/allowedrepos/samples/repo-must-be-openpolicyagent command. If we run a 
+
+```bash
+kubectl get pods
+``` 
+
+we should see the following output:
+
+```bash
+NAME                  READY   STATUS    RESTARTS   AGE
+opa-allowed           1/1     Running   0          76s
+```
+
+If we inspect the pods and look at the containers.spec section of the pod we see the following:
+
+```yaml
+Containers:
+  opa:
+    Container ID:  docker://1b2da3c7d7c41becac49613ed7db863c7b1365137bbfe1b108220d4ffba188b3
+    Image:         openpolicyagent/opa:0.9.2
+    Image ID:      docker-pullable://openpolicyagent/opa@sha256:04ff8fce2afd1a3bc26260348e5b290e8d945b1fad4b4c16d22834c2f3a1814a
+    Port:          <none>
+    Host Port:     <none>
+    Args:
+      run
+      --server
+      --addr=localhost:8080
+```
+
+This is exactly what is defined in our example_allowed.yaml file so we know that our policy was deployed successfully using OPA Gatekeeper. For more information on OPA Gatekeeper policies, check out the GitHub repo which can be found [here](https://github.com/open-policy-agent/gatekeeper-library)
```

Comment:
We need to show an example how customers will apply OPA policies with SSP. Where will those policies be located? Let's say we need to set cluster level or namespace level policies. The platform admin wants them to be in place when we provision clusters. While the examples are great, the practical aspects, e.g. an example solution is still missing. Should it be through the blueprint? GitOps? If so, how?